crates/gossip/src/gossip.rs

```diff
@@ -698,17 +642,8 @@ mod test {
             .await
             .unwrap();
 
-        harness_1
```

Comment:
This should be necessary shouldn't it? I don't think anything else would do the update of the DHT model and if it's not updated then the DHT diff will just return "no diff" and not sync anything

Reply:
It's not accessible and the test passed without it. Right, then something must be wrong.

Reply:
Let me check this out and run the tests. It should be possible to see from the logs which sync method it's choosing. I'd like to be able to assert but what it did isn't recorded anywhere yet

Reply:
Okay, go ahead. I'm happy to do it as well. Sorry I castrated the tests. I was mostly focused on the transport issue.

Reply:
Haha! No worries, they're not the best tests because it's not entirely clear what they do without an assertion at the end that they've used a particular mechanism to achieve something. I will look for ways to improve that but I've provided some suggested solutions now